indigo_api/models/works.py

```diff
@@ -243,10 +257,8 @@ def commenceable_provisions(self):
             plugin = plugins.for_document('toc', doc)
             if plugin:
                 toc = plugin.table_of_contents_for_document(doc)
-                for item in plugin.commenceable_items(toc):
-                    if item.id and item.id not in id_set:
-                        id_set.add(item.id)
-                        provisions.append(item)
+                items = plugin.commenceable_items(toc)
+                provisions = self.insert_provisions(provisions, id_set, items)
```

Comment:
you're going to have to walk me through what's going on here. 

i'm also not sure that `insert_provisions` is a useful helper method on the work object.

Reply:
ah, I get it. two things:

1. `insert_provisions` is a very specific helper function. i don't think it has external value. i suggest just pulling that loop into the `commenceable_provisions`
2. some additional comments on what `insert_provisions` is doing would make it a bit clearer. I've added some suggestions.

Reply:
Oh, I remember now. I moved `insert_provisions` out so I could test it. What's the better way of doing that?

Reply:
That's a good reason. Can we move lines 259 onwards into the TOC plugin class? I think it makes more sense to have the helper function there, which  makes it easier to test. Something like:

```python
plugin.insert_commenceable_provisions(doc, provisions, id_set)
```

and the plugin then splits out those two functions. It doesn't need to return anything because `provisions` and `id_set` are updated in-place.

Reply:
Great idea, thanks!